npm_and_yarn/helpers/lib/npm/vulnerability-auditor.js

```diff
@@ -0,0 +1,207 @@
+/* Vulnerability auditor for npm
+ *
+ * Inputs:
+ *  - path to directory containing a package.json and a package-lock.json
+ *  - array of security advisories to audit for,
+ *      [
+ *        { dependency_name: string, affected_versions: [string, ...] },
+ *        ...
+ *      ]
+ *
+ * Outputs:
+ *  - object indicating whether a fix is available and how to achieve it,
+ *      {
+ *        dependency_name: string,
+ *        current_version: string,
+ *        target_version: string,
+ *        fix_available: boolean | object,
+ *        fix_updates: [
+ *          {
+ *            dependency_name: string,
+ *            current_version: string,
+ *            target_version: string
+ *          },
+ *          ...
+ *        ]
+ *      }
+ */
+
+const Arborist = require('@npmcli/arborist')
+const nock = require('nock')
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec)
+
+async function findVulnerableDependencies(directory, advisories) {
+  const npmConfig = await loadNpmConfig()
+  const caCerts = loadCACerts(npmConfig)
```

Comment:
Would be neat if there was lib or something in npm that could be re-used to load the config for arborist? Can't see this but maybe @nlf or @wraithgar might know?

Reply:
This is a known pain point on the cli, and it's due to the fact that the config is functionally split between the cli itself (where config definitions live) and `@npmcli/config` (where the config logic itself lives).  There is no better way today to do this in tests right now, but it's on our roadmap for improvement.

Until config is pulled out into a more reproducible format, getting it straight from the npm cli is absolutely the best place to build from.

Reply:
@wraithgar thanks for confirming—we weren't sure if this was the best way to do this. We did look at pulling in `@npmcli/config` but we felt a lot of friction in trying to get it configured correctly so ultimately we chose this.

Reply:
It looks like we'll also need to populate any scoped/global registry settings from the npm config here to support private registries.

Reply:
I followed up with populating registry settings from the npm config in https://github.com/dependabot/dependabot-core/pull/5313. This ended up being a bit awkward as to get the credentials we needed to bypass the npm cli's result (which omit credentials) and read them directly from `.npmrc`